src/vivarium/framework/population/manager.py

```diff
@@ -399,19 +393,114 @@ def _create_simulants(
+                    f"returned a {type(tracked)}."
+                )
+            idx = tracked[tracked == True].index
+
+        if isinstance(attributes, list):
+            # check for duplicate request:
+            duplicates = list(set([x for x in attributes if attributes.count(x) > 1]))
+            if duplicates:
+                raise PopulationError(f"Duplicate attributes requested: {duplicates}")
+
+        attributes_to_include = (
+            self._attribute_pipelines.keys() if attributes == "all" else attributes
+        )
+
+        non_existent_attributes = set(attributes_to_include) - set(self._attribute_pipelines)
+        if non_existent_attributes:
+            raise PopulationError(
+                f"Requested attribute(s) {non_existent_attributes} not in population table. "
+                "This is likely due to a failure to require some columns, randomness "
+                "streams, or pipelines when registering a simulant initializer, a value "
+                "producer, or a value modifier. NOTE: It is possible for a run to "
+                "succeed even if resource requirements were not properly specified in "
+                "the simulant initializers or pipeline creation/modification calls. This "
+                "success depends on component initialization order which may change in "
+                "different run settings."
+            )
+
+        attributes_list: list[pd.Series[Any] | pd.DataFrame] = []
+
+        # batch simple attributes and pop right off the backing data
+        simple_attributes = [
+            name
+            for name, pipeline in self._attribute_pipelines.items()
+            if name in attributes_to_include and pipeline.is_simple
+        ]
+        if simple_attributes:
+            attributes_list.append(self._population.loc[idx, simple_attributes])
+
+        # handle remaining non-simple attributes one by one
+        remaining_attributes = [
+            attribute
+            for attribute in attributes_to_include
+            if attribute not in simple_attributes
+        ]
+        for name in remaining_attributes:
+            pipeline = self._attribute_pipelines[name]
+            values = self._population.loc[idx, name] if pipeline.is_simple else pipeline(idx)
+            if isinstance(values, pd.Series):
+                values.name = name
+            attributes_list.append(values)
+
+        df = (
+            pd.concat(attributes_list, axis=1) if attributes_list else pd.DataFrame(index=idx)
+        )
+
+        duplicate_columns = df.columns[df.columns.duplicated()].tolist()
+        if duplicate_columns:
+            raise PopulationError(
+                f"Population table has duplicate column names: {duplicate_columns}. "
+                "This is likely due to an AttributePipeline producing a pd.Dataframe with the "
+                "same column name(s) that some Component has in its `columns_created` property."
+            )
+
+        return df
```

Comment:
One note is we've lost control of the column order at this point. Does that actually matter?

It's tricky to handle since a single attribute could return multiple cols so it's not as easy as just returning `df[attributes_to_include]`

Reply:
After discussion:
- try to preserve order
- prepend dataframe pipelines with the pipeline name, e.g. `acceleration.x`
- look into multi-indexes instead. We def need to support "usual" behavior though, so if a mutli-index messes that up then it's a no-go

Reply:
Do this all as a follow-on pr